File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/SubstancePolymer.ts
import { Attachment } from "./Attachment";
import { CodeableConcept } from "./CodeableConcept";
import { SubstanceAmount } from "./SubstanceAmount";
import { DomainResource } from "./DomainResource";
import { BackboneElement } from "./BackboneElement";

export type SubstancePolymerRepeatRepeatUnitDegreeOfPolymerisation = BackboneElement & {
    amount?: SubstanceAmount;
    degree?: CodeableConcept;
};

export type SubstancePolymerRepeatRepeatUnitStructuralRepresentation = BackboneElement & {
    type?: CodeableConcept;
    attachment?: Attachment;
    representation?: string;
};

export type SubstancePolymerRepeatRepeatUnit = BackboneElement & {
    amount?: SubstanceAmount;
    repeatUnit?: string;
    degreeOfPolymerisation?: SubstancePolymerRepeatRepeatUnitDegreeOfPolymerisation[];
    structuralRepresentation?: SubstancePolymerRepeatRepeatUnitStructuralRepresentation[];
    orientationOfPolymerisation?: CodeableConcept;
};

export type SubstancePolymerRepeat = BackboneElement & {
    repeatUnit?: SubstancePolymerRepeatRepeatUnit[];
    numberOfUnits?: number;
    repeatUnitAmountType?: CodeableConcept;
    averageMolecularFormula?: string;
};

export type SubstancePolymerMonomerSetStartingMaterial = BackboneElement & {
    type?: CodeableConcept;
    amount?: SubstanceAmount;
    material?: CodeableConcept;
    isDefining?: boolean;
};

export type SubstancePolymerMonomerSet = BackboneElement & {
    ratioType?: CodeableConcept;
    startingMaterial?: SubstancePolymerMonomerSetStartingMaterial[];
};

export type SubstancePolymer = DomainResource & {
    class_?: CodeableConcept;
    repeat?: SubstancePolymerRepeat[];
    geometry?: CodeableConcept;
    monomerSet?: SubstancePolymerMonomerSet[];
    modification?: string[];
    copolymerConnectivity?: CodeableConcept[];
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/SubstancePolymer.py
from typing import Optional, List
from pydantic import *
from ..base import *

class SubstancePolymer_Repeat_RepeatUnit_DegreeOfPolymerisation(BackboneElement):
    amount: Optional[SubstanceAmount] = None
    degree: Optional[CodeableConcept] = None

class SubstancePolymer_Repeat_RepeatUnit_StructuralRepresentation(BackboneElement):
    type: Optional[CodeableConcept] = None
    attachment: Optional[Attachment] = None
    representation: Optional[str] = None

class SubstancePolymer_Repeat_RepeatUnit(BackboneElement):
    amount: Optional[SubstanceAmount] = None
    repeat_unit: Optional[str] = None
    degree_of_polymerisation: Optional[List[SubstancePolymer_Repeat_RepeatUnit_DegreeOfPolymerisation]] = None
    structural_representation: Optional[List[SubstancePolymer_Repeat_RepeatUnit_StructuralRepresentation]] = None
    orientation_of_polymerisation: Optional[CodeableConcept] = None

class SubstancePolymer_Repeat(BackboneElement):
    repeat_unit: Optional[List[SubstancePolymer_Repeat_RepeatUnit]] = None
    number_of_units: Optional[int] = None
    repeat_unit_amount_type: Optional[CodeableConcept] = None
    average_molecular_formula: Optional[str] = None

class SubstancePolymer_MonomerSet_StartingMaterial(BackboneElement):
    type: Optional[CodeableConcept] = None
    amount: Optional[SubstanceAmount] = None
    material: Optional[CodeableConcept] = None
    is_defining: Optional[bool] = None

class SubstancePolymer_MonomerSet(BackboneElement):
    ratio_type: Optional[CodeableConcept] = None
    starting_material: Optional[List[SubstancePolymer_MonomerSet_StartingMaterial]] = None

class SubstancePolymer(DomainResource):
    class: Optional[CodeableConcept] = None
    repeat: Optional[List[SubstancePolymer_Repeat]] = None
    geometry: Optional[CodeableConcept] = None
    monomer_set: Optional[List[SubstancePolymer_MonomerSet]] = None
    modification: Optional[List[str]] = None
    copolymer_connectivity: Optional[List[CodeableConcept]] = None